Sources/Web3Core/KeystoreManager/AbstractKeystore.swift

```diff
@@ -11,11 +11,30 @@ public protocol AbstractKeystore {
     func UNSAFE_getPrivateKeyData(password: String, account: EthereumAddress) throws -> Data
 }
 
-public enum AbstractKeystoreError: Error {
-    case noEntropyError
-    case keyDerivationError
-    case aesError
-    case invalidAccountError
+public enum AbstractKeystoreError: LocalizedError {
+    case noEntropyError(_ additionalDescription: String? = nil)
+    case keyDerivationError(_ additionalDescription: String? = nil)
+    case aesError(_ additionalDescription: String? = nil)
+    case invalidAccountError(_ additionalDescription: String? = nil)
     case invalidPasswordError
-    case encryptionError(String)
+    case encryptionError(_ additionalDescription: String? = nil)
+
+    public var errorDescription: String? {
+        var errorMessage: [String?]
+        switch self {
+        case .noEntropyError(let additionalDescription):
+            errorMessage = ["Entropy error (e.g. failed to generate a random array of bytes).", additionalDescription]
```

Comment:
I wonder if this line is a cause that we're messing here with an array of Strings rather a plain String? Here it looks quite more necessary than in the previous alike code snippet, yet I wonder if it's a best way in terms of code intention clarity to implement such behaviour?